akka-projection-r2dbc/src/main/scala/akka/projection/r2dbc/internal/R2dbcProjectionImpl.scala

```diff
@@ -362,6 +363,17 @@ private[projection] object R2dbcProjectionImpl {
                 Some(loadedEnvelope)
               }
             } else {
+              // FIXME do we need the same for all projection handler variants?
+              // FIXME not accepted may be that a producer filter decided to start including events,
+              //       how do we know for sure and what offset to start from?
+              env match {
+                case ee: EventEnvelope[_] if ee.sequenceNr > 0 =>
+                  log.debug("Triggering replay for unexpected offset envelope: [{}]", env)
+                  ConsumerFilter(system).ref ! ConsumerFilter.Replay(
+                    "producer-filter-e2e-test", // FIXME where do we get stream id from?
+                    Set(ConsumerFilter.EntityIdOffset(ee.persistenceId, 0L)))
```

Comment:
That should be `1L` if the purpose is to replay all events.
However, the offset store knows what seqNr it has processed so the replay should start from that seqNr instead of replaying all.

Reply:
Got it